### PR TITLE
Action lint workflow

### PR DESCRIPTION
**Pull request by stevenhorsman**
On PRs that update anything in the workflows directory, add an actionlint run to validate our workflow files for errors and hopefully catch issues earlier.

As part of this PR that means also fixing the ~120 linter errors we already have!

Fixes: #9646

### CODE REVIEW

**Comment by stevenhorsman** [comment]
So the new workflow has run successfully (https://github.com/kata-containers/kata-containers/actions/runs/12198124056/job/34029188214?pr=10631) to show we don't have lints, but across such a wide range of workflow changes I'm not sure there is a good way to test that no behaviour has changed before this is merged?

**Review by fidencio — APPROVED** [review]
lgtm, thanks @stevenhorsman!

**Comment by stevenhorsman** [comment]
Adding `do-not-merge`, which I test out the nightly on this code: https://github.com/kata-containers/kata-containers/actions/runs/12198210761

File: .github/workflows/build-kata-static-tarball-ppc64le.yaml [hunk 1]
```diff
@@ -167,8 +167,8 @@ jobs:
     steps:
       - name: Prepare the self-hosted runner
         run: |
-            ${HOME}/scripts/prepare_runner.sh
-            sudo rm -rf $GITHUB_WORKSPACE/*
+            "${HOME}/scripts/prepare_runner.sh"
+            sudo rm -rf "$GITHUB_WORKSPACE/*"
```
Comment (wainersm) [inline .github/workflows/build-kata-static-tarball-ppc64le.yaml:171]: Hi @stevenhorsman ,

I think here and in subsequent occurrences you want `"$GITHUB_WORKSPACE"/*`.

```
$ GITHUB_WORKSPACE=/tmp; ls "$GITHUB_WORKSPACE/*"
ls: cannot access '/tmp/*': No such file or directory
$ GITHUB_WORKSPACE=/tmp; ls "$GITHUB_WORKSPACE"/*
/tmp/foo.txt                 /tmp/qtsingleapp-zoom-18f2b-lockfile  /tmp/values_yaml.BkUvIZ  /tmp/values_yaml.ZRfAp3
/tmp/qtsingleapp-zoom-18f2b  /tmp/values_yaml.9n6WOV               /tmp/values_yaml.Qv6f9B
...
...
```

BTW, `sudo rm -rf "$GITHUB_WORKSPACE/*"` is extremely dangerous. I don't know why shellchek doesn't complain. Maybe it does and is fixed on one of the following commits? :)

Comment (stevenhorsman) [reply .github/workflows/build-kata-static-tarball-ppc64le.yaml:171]: Thanks for spotting this. I made that mistake elsewhere and thought I'd corrected them. I will do now. Thanks for the review. Any no shellcheck doesn't complain

**Review by wainersm — APPROVED** [review]
Anyone fixing the lint warnings of our workflows deserves a honor medal! Thanks @stevenhorsman !

**Comment by stevenhorsman** [comment]
This is a workflow only change that isn't tested by the PR checks as it is not triggered with a on/pull_request. It also has at least two reviews with the force-skip-ci label, therefore we can skip the CI and use admin rights to merge this.

**Pull request merged**